### ISSUE TITLE
[V2][iOS] topBar Large title flickering while transition on back button press and color problem

### ISSUE THREAD

**Issue by Crisis2010** [wix/react-native-navigation#5145]
### Issue Description

Here are a few problems that I found with **iOS Large Title**:

1. **Navigation.setDefaultOptions** not working with Large Title color, this only works if set color in every **Navigation.setRoot children**

2. When you go back by pressing **back button** to the page with Large Title, Search Bar and custom backgroundColor, animation is flickering or more like backgroundColor  behind is transparent and not changed to custom color. Also Large title color is black while transition. This is **ONLY** if going back by left top **back button**, and **ok** if going back by **popGesture**

### Steps to Reproduce / Code Snippets / Screenshots
Going back by **Button** _(bug)_
![bug](https://i.ibb.co/FxLtYTD/2019-05-23-19-47-20.gif)

Going back by **Gesture** _(ok)_
![bug](https://i.ibb.co/Bw8qp7x/2019-05-23-19-49-48.gif)

My DefaultOptions code (largeTitle color not working)
```js 
  Navigation.setDefaultOptions({
    topBar: {
      background: {
        color: '#30291E',
      },
      title: {
        color: 'white',
        largeTitle: {
          fontSize: 34,
          color: 'white', // <---- Not work
        },
      },
      backButton: {
        title: '',
        color: 'white',
      },
      buttonColor: 'white',
    },
  });
 ```

I think it supposed to be fixed, but now it could be fixed by, but while transition it still black
```js 
  Navigation.setRoot({
    root: {
      bottomTabs: {
        children: [
          // (...),
          {
            stack: {
              children: [
                {
                  component: {
                    name: CATEGORY_SCREEN,
                    options: {
                      topBar: {
                        searchBar: true,
                        searchBarHiddenWhenScrolling: true,
                        searchBarPlaceholder: 'Search',
                        largeTitle: {
                          visible: true,
                          color: 'white',  // <--- FIX
                        },
                        title: {
                          text: 'Catalog',
                        },
                      },
                    },
                  },
                },
              ],
            },
          },
        ],
      },
    },
  });
 ```
---
### Environment
* React Native Navigation version: 2.20.1
* React Native version: 0.59.8
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): Device and Simulator, Debug and Release

**Comment by Crisis2010** [comment]
Still exist

**Comment by Crisis2010** [comment]
And there is even more bugs with also searchBar enabled.
Going back by **Button** and **Gesture** _(bug)_
![bug](https://i.ibb.co/JqvPSW8/2019-06-24-09-45-51.gif)

Going back by **Button** and **Gesture** _(bug)_
![bug](https://i.ibb.co/Q8GcM3V/2019-06-24-09-46-21.gif)

In both case on both screens enabled largeTitle and searchBar.

**Comment by hahtml** [comment]
Same issue here.
That flickering happens randomly with both large title and search bar enabled.

**Comment by Crisis2010** [comment]
Not stale, still hope it will be fixed in 3.0 #5228

**Comment by t3hfx** [comment]
Faced the same problem with going back by button from the larger title to original topBar. Trying to figure out why is it happening
